datafusion/physical-plan/src/coalesce_batches.rs

```diff
@@ -412,4 +430,100 @@ mod tests {
         )
         .unwrap()
     }
+
+    #[test]
+    fn test_gc_string_view_batch_small_no_compact() {
+        // view with only short strings (no buffers) --> no need to compact
+        let array = StringViewTest {
+            rows: 1000,
+            strings: vec![Some("a"), Some("b"), Some("c")],
+        }
+        .build();
+
+        let gc_array = do_gc(array.clone());
+        compare_string_array_values(&array, &gc_array);
+        assert_eq!(array.data_buffers().len(), 0);
+        assert_eq!(array.data_buffers().len(), gc_array.data_buffers().len()); // no compaction
+    }
+
+    #[test]
+    fn test_gc_string_view_batch_large_no_compact() {
+        // view with large strings (has buffers) but full --> no need to compact
+        let array = StringViewTest {
+            rows: 1000,
+            strings: vec![Some("This string is longer than 12 bytes")],
+        }
+        .build();
+
+        let gc_array = do_gc(array.clone());
+        compare_string_array_values(&array, &gc_array);
+        assert_eq!(array.data_buffers().len(), 5);
+        // TODO this is failing now (it always compacts)
```

Comment:
This test is failing now because the StringView is compacted even though all data in the buffers is pointed to

What I think this is showing is that string views with many long strings but no actual garbage are always compacted, which doesn't seem like a good heuristic

Reply:
I agree the heuristic is a bit arbitrary and not really backed by real experiments... I'll think a bit more about this and back to it later today

Reply:
FWIW I am going to prototype what the combined filter/coalsece thing we were talking about on https://github.com/apache/datafusion/issues/11628 might look like

Reply:
Prototype is here: https://github.com/apache/datafusion/pull/11647 (not yet ready for review)